consensus/state_sync/module.go

```diff
@@ -40,25 +55,83 @@ func (*stateSync) Create(bus modules.Bus, options ...modules.ModuleOption) (modu
 	bus.RegisterModule(m)
 
 	m.logger = logger.Global.CreateLoggerForModule(m.GetModuleName())
+	m.committedBlocksChannel = make(chan uint64, committedBlocsChannelSize)
 
 	return m, nil
 }
 
-// TODO(#352): implement this function
-// Start performs state sync
+func (m *stateSync) Set(aggregatedMetaData *typesCons.StateSyncMetadataResponse) {
+	m.aggregatedMetaData = aggregatedMetaData
+}
+
+func (m *stateSync) StartSyncing() {
+	err := m.Start()
+	if err != nil {
+		m.logger.Error().Err(err).Msg("couldn't start state sync")
+	}
+}
+
+// Start performs state sync process, starting from the consensus module's current height to the aggragated metadata height
 func (m *stateSync) Start() error {
-	// processes and aggregates all metadata collected in metadataReceived channel,
-	// requests missing blocks starting from its current height to the aggregated metadata's maxHeight,
-	// once the requested block is received and committed by consensus module, sends the next request for the next block,
-	// when all blocks are received and committed, stops the state sync process by calling its `Stop()` function.
-	return nil
+	consensusMod := m.bus.GetConsensusModule()
+	currentHeight := consensusMod.CurrentHeight()
+	nodeAddress := consensusMod.GetNodeAddress()
+	readCtx, err := m.GetBus().GetPersistenceModule().NewReadContext(int64(currentHeight))
+	if err != nil {
+		return err
+	}
+	defer readCtx.Release()
+
+	// get the current validators
+	m.validators, err = readCtx.GetAllValidators(int64(currentHeight))
+	if err != nil {
+		return err
+	}
+
+	// requests blocks from the current height to the aggregated metadata height
+	for currentHeight <= m.aggregatedMetaData.MaxHeight {
```

Comment:
So this whole function is blocking?

I thought we wanted a background channel that always does this asynchronously, no?

Reply:
added a background channel process for passive state sync. Active state sync is still like this though, added a note to consider changing this.